### PR TITLE
Added spec changes to Machinepool and verified codegen

### PR DESCRIPTION
**Pull request by bkhadars**

### CODE REVIEW

File: pkg/types/powervs/machinepools.go [hunk 1]
```diff
@@ -20,11 +20,11 @@ type MachinePool struct {
 
 	// Memory defines the memory in GB for the instance.
 	//
-	Memory int `json:"memory"`
+	Memory string `json:"memory"`
```
Comment (mjturek) [inline pkg/types/powervs/machinepools.go:23]: Just curious, why are we switching to string?

File: pkg/types/powervs/machinepools.go [hunk 2]
```diff
@@ -20,11 +20,11 @@ type MachinePool struct {
...
 
 	// Processors defines the processing units for the instance.
 	// @TODO:
-	Processors float32 `json:"processors"`
+	Processors string `json:"processors"`
```
Comment (mjturek) [inline pkg/types/powervs/machinepools.go:27]: ditto

**Comment by mjturek** [comment]
Questions inline

**Comment by clnperez** [comment]
Answers resolved offline, and I think I had already fixed these things but not backported, so, merging. Thanks @bkhadars!

**Pull request merged**